src/HonzaBotner.Discord.Services/EventHandlers/ReminderReactionsHandler.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Threading.Tasks;
+using DSharpPlus.Entities;
+using DSharpPlus.EventArgs;
+using HonzaBotner.Discord.EventHandler;
+using HonzaBotner.Discord.Services.Options;
+using HonzaBotner.Services.Contract;
+using Microsoft.Extensions.Options;
+
+namespace HonzaBotner.Discord.Services.EventHandlers
+{
+    public class ReminderReactionsHandler : IEventHandler<MessageReactionAddEventArgs>
+    {
+        private readonly ReminderOptions _options;
+
+        private readonly IRemindersService _service;
+
+        public ReminderReactionsHandler(IOptions<ReminderOptions> options, IRemindersService service)
+        {
+            _options = options.Value;
+            _service = service;
+        }
+
+        public async Task<EventHandlerResult> Handle(MessageReactionAddEventArgs arguments)
+        {
+            // TODO: Move this to framework level code
+            if (arguments.User.IsBot)
+            {
+               return EventHandlerResult.Stop;
+            }
+
+            var emoji = arguments.Emoji;
+            var reminder = await _service.GetByMessageIdAsync(arguments.Message.Id);
+
+            if (reminder == null)
+            {
+                return EventHandlerResult.Continue;
+            }
+
+            // The owner has cancelled the reminder
+            if (emoji == _options.CancelEmojiName && arguments.User.Id == reminder.OwnerId)
+            {
+                await _service.DeleteReminderAsync(reminder);
+                await arguments.Message.ModifyAsync("", CreateCancelledReminderEmbed());
+                await arguments.Message.DeleteAllReactionsAsync("Reminder cancelled");
+
+                return EventHandlerResult.Stop;
+            }
+
+            // Somebody else wants to be mentioned within the reminder notification
+            if (emoji == _options.JoinEmojiName && arguments.User.Id != reminder.OwnerId)
+            {
+                // There is nothing that has to be done, as the reactions are checked during the reminder notification
+                // This check is just to disable owner from joining his own reminder
```

Comment:
This logic is little bit weird, or at least sounds a bit dangerous.

Reply:
Elaborate pls

Reply:
Seems ok to me. The bot doesn't have to care about people reacting. It only handles the reactions at the moment when the reaction triggers its time.